[FEATURE] Dans Pix App, déconnecter un utilisateur après qu'il a envoyé ses résultats d'un parcours simplifié (PIX-2011).

chrc opened the pull request:
## :unicorn: Problème
Dans Pix App, à l'issue d'un parcours simplifié, et après avoir envoyé ses résultats, l'utilisateur reste connecté.

## :robot: Solution
* Si c'est un parcours simplifié, après l'envoie des résultats, forcer la déconnexion,
et rediriger vers la page du site vitrine pix.fr.

## :rainbow: Remarques
*Si l'environnement n'est pas PROD, alors la redirection sera faite vers la page de connexion.

## :100: Pour tester
* Se connecter à Pix App
* Commencer un parcours avec accès simplifié (avec le code campagne SIMPLIFIE en RA)
* A l'issue de ce parcours, envoyer les résultats
* Constater que la redirection vers le site vitrine* est bien faite (la session est bien supprimée).

_Relancer les seeds si besoin  :_
```shell
scalingo --region osc-fr1 --app pix-api-review-pr2427 run bash
npm run db:seed
```

chrc committed:
[APP] Add disconnectUser method in skill-review

pix-service commented:
I'm deploying this PR to these urls:

- App: https://app-pr2427.review.pix.fr
- Orga: https://orga-pr2427.review.pix.fr
- Certif: https://certif-pr2427.review.pix.fr
- Admin: https://admin-pr2427.review.pix.fr
- API: https://api-pr2427.review.pix.fr/api/

 Please check it out!

thloesch commented on mon-pix/app/components/routes/campaigns/assessment/skill-review.js:
```diff
@@ -51,8 +54,13 @@ export default class SkillReview extends Component {
+      .then(async () => {
         campaignParticipation.isShared = true;
+
+        const isSimplifiedAccess = await this.args.model.campaignParticipation.campaign.get('isSimplifiedAccess');
```
**Suggestion non-bloquante**

Toute petite suggestion pour éliminer les éventuelles incompréhensions dans ce contexte

```suggestion
        const isSimplifiedAccessCampaign = await this.args.model.campaignParticipation.campaign.get('isSimplifiedAccess');
```

chrc replied:
Done

thloesch commented:
J'ai du mal à tester fonctionnellement en RA.

J'ai rajouté en base un target profile et les target-profile-skills associés, créé une campagne avec ce target profile, puis j'ai passé le parcours de cette campagne avec plusieurs utilisateurs.
À chaque essai, lorsque je clique sur envoyer mes résultats je me retrouve avec une erreur

![Capture d’écran 2021-01-21 à 14 50 36](https://user-images.githubusercontent.com/46494361/105360072-7a7d9680-5bf8-11eb-8e10-74fcea0fcf5e.png)

En cliquant sur le bouton "Reprendre", j'arrive directement ici

![Capture d’écran 2021-01-21 à 14 50 49](https://user-images.githubusercontent.com/46494361/105360121-8a957600-5bf8-11eb-9029-6cb2a20579f4.png)

Et lorsque je clique sur continuer, je suis redirigé vers mon profil, et donc sans être déconnecté.

chrc committed:
[API] Add simplified access seeds

Add an simplified access target-profile seed.
Add an simplified access campaign seed.

thloesch commented:
Revue fonctionnelle OK ✅

thloesch commented on api/db/seeds/data/target-profiles-builder.js:
```diff
@@ -107,4 +107,31 @@ module.exports = function targetProfilesBuilder({ databaseBuilder }) {
+  databaseBuilder.factory.buildCampaign({
+    targetProfileId: simplifiedAccesTargetProfile.id,
+    organizationId: 1,
+    code: 'SIMPLIFIE',
```
**Remarque** Pour retrouver plus facilement ce code campagne, je pense qu'il est préférable de déplacer ce code dans les seeds des campagnes. Quitte à y rajouter la création du target profile associé.

Ultimement, ce qui nous intéresse c'est d'avoir facilement un parcours avec accès simplifié.

chrc replied:
Done

thloesch commented on api/db/seeds/data/campaigns-builder.js:
```diff
@@ -128,4 +128,12 @@ module.exports = function campaignsBuilder({ databaseBuilder }) {
+    code: 'SIMPLIFIE',
+    name: 'Parcours simplifié',
+    organizationId: 1,
+    targetProfileId: 100322,
```
**BSR non-bloquant**

Super le déplacement dans le campaign-builder 👍

Du coup je me rends compte qu'il y a pas mal de magic numbers dans ce fichier, surtout autour des IDs (orga et target profiles utilisés).
On pourrait faire la modif dans cette PR ou l'isoler dans un ticket à part. C'est peut-être un sujet qui mérite du temps dédié.

chrc replied:
Dans une autre PR, car sujet transverse à toutes les teams : Comment gérer les IDs cross seeds ?

octo-topi commented on mon-pix/app/components/routes/campaigns/assessment/skill-review.js:
```diff
@@ -68,4 +76,9 @@ export default class SkillReview extends Component {
     return this.router.transitionTo('campaigns.start-or-resume', assessment.codeCampaign);
   }
 
+  async disconnectUser() {
```
**chouette**
j'aime cette expression d'intention !

octo-topi commented on mon-pix/tests/unit/components/routes/campaigns/skill-review-test.js:
```diff
@@ -41,16 +42,29 @@ describe('Unit | component | Campaigns | Evaluation | Skill Review', function()
       expect(component.args.model.campaignParticipation.isShared).to.equal(true);
     });
+
+    it('should disconnect user if campaign has simplified access', async function() {
```
**remarque**
il n'y a pas de test d'intégration ni d'acceptance, parce que la déconnexion réelle décharge de la mémoire la lib de test, et donc le test est impossible

chrc replied:
Mais dans l'absolu, je voudrais bien creuser un helper/addon pour pouvoir tester en acceptance (ou e2e) ces cas de figure :)
